### ISSUE TITLE
Code coverage is broken

### ISSUE THREAD

**Issue by josdejong** [josdejong/mathjs#2871]
When running code coverage via `npm run coverage`, none of the source code files is covered except for two CommonJS files, I think this is broken since we migrated the code from CommonJS to ESM.

The generated report `./coverage/lcov-report/index.html` looks as follows, only scanning `tools/approx.js` and `tools/utils.js`. Instead, it should show scan results of hundreds of files.

![afbeelding](https://user-images.githubusercontent.com/568626/210331587-0b38cd07-bfca-4244-ace8-ed9777941166.png)

To solve this we can:

- see if we we can configure the coverage tool `nyc` to also scan ESM files
- replace the coverage tool with an alternative

Help to get this fixed would be very welcome!

**Comment by bornova** [comment]
I would suggest replacing `nyc` with [c8](https://github.com/bcoe/c8).  Works great with ESM and compatible with `Istanbul`'s reporters.

```
npm un Istanbul nyc
```
```
npm i c8 --save-dev
```

and change in package.json:
```json
"coverage": "nyc --reporter=lcov --reporter=text-summary mocha test/unit-tests && echo \"\nDetailed coverage report is available at ./coverage/lcov-report/index.html\"",
```
to
```json
"coverage": "c8 --reporter=lcov --reporter=text-summary mocha test/unit-tests && echo \"\nDetailed coverage report is available at ./coverage/lcov-report/index.html\"",
```

Seems to work good on my machine:
![image](https://user-images.githubusercontent.com/35872220/211177371-0726ca9b-0219-48ad-be8d-468411d05010.png)

**Comment by josdejong** [comment]
Fix is published now in `v11.5.1`.
